app/apps/workflow/models.py

```diff
@@ -132,76 +138,83 @@ def wait_for_workflows_and_send_message(message):
-                messages = main_workflow.data.get("messages", [])
+                extra_data = {
+                    "all_summons": all_summons,
+                }
 
-                # add message to main workflow, this will tell only the main workflow to go on
-                messages.append(message)
-                main_workflow.data.update(
-                    {
-                        "messages": messages,
-                        "all_summons": all_summons,
-                    }
-                )
-                main_workflow.save(update_fields=["data"])
-                main_workflow.update_workflow()
+                accept_message_for_workflow.delay(main_workflow.id, message, extra_data)
 
                 # delete others
-                other_workflows = all_workflows.exclude(id=main_workflow.id)
-                other_workflows.delete()
+                # other_workflows = all_workflows.exclude(id=main_workflow.id)
+                # other_workflows.delete()
+
+        def start_subworkflow(subworkflow_name):
+            print(f"subworkflow name: {subworkflow_name}")
+            subworkflow = CaseWorkflow.objects.create(
+                case=case,
+                workflow_type=subworkflow_name,
+            )
+            subworkflow.set_initial_data(workflow_instance.get_data())
 
         wf.script_engine = BpmnScriptEngine(
             scriptingAdditions={
                 "set_status": set_status,
                 "wait_for_workflows_and_send_message": wait_for_workflows_and_send_message,
+                "start_subworkflow": start_subworkflow,
             }
         )
         return wf
 
-    def _print_task_data(self, wf):
-        print("START: print data")
-        print(wf.name)
-        wf.dump()
-        for t in wf.get_tasks():
-            print("-")
-            print(t.get_state_name())
-            print(t.workflow.name)
-            print(t.task_spec.name)
-            print(t.last_state_change)
-            print(t.state_history)
-            print(t.log)
-            print(t.data)
-            print(t.task_info())
-        print("END: print data")
-
     def get_user_task_form(self, task_id):
         wf = self.get_or_restore_workflow_state()
         task = wf.get_task(task_id)
         if task:
-            return Task.parse_task_spec_form(task.task_spec.form)
+            return CaseUserTask.parse_task_spec_form(task.task_spec.form)
         return []
 
     def message(self, message_name, payload, resultVar, extra_data={}):
         wf = self.get_or_restore_workflow_state()
-
+        data = {}
+        data.update(extra_data)
+        self.set_initial_data(extra_data)
         wf = self._update_workflow(wf)
         wf = self._message(wf, message_name, payload, resultVar, extra_data)
-
         wf = self._update_workflow(wf)
         self.save_workflow_state(wf)
         self._update_db(wf)
 
-    def accept_message(self, message_name):
+    def accept_message(self, message_name, extra_data):
```

Comment:
Why all the print statements? Can we use a logger instead?

Reply:
Local shit, left out in next pr